### PR TITLE
Update `clone()` for pipelines and components to copy over random state by default

### LINKED ISSUE #1340: Should clone copy the random state by default?
In #1321, we had to modify the pipeline.clone call in `_compute_cv_scores` to pass in the random state of the automl search object - without this the pipeline always get init with random state 0. 

Comments from the original PR:

Yep

RE your comment in standup this week: @eccabay and I chose this behavior because we felt copying random_state under the hood was dirty. However, in light of our recent conversation about #1324 (component/pipeline determinism), perhaps we should revisit this, and have `clone` copy `random_state` by default. Do you have thoughts @freddyaboulton @eccabay ? We can move the discussion to an issue and resolve there.

_Originally posted by @dsherry in https://github.com/alteryx/evalml/pull/1321#discussion_r510911430_

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1340

### CODE REVIEW

File: evalml/tests/component_tests/test_components.py [hunk 1]
```diff
@@ -440,27 +440,19 @@ def test_clone_init():
     clf = MockFitComponent(**params)
     clf_clone = clf.clone()
     assert clf.parameters == clf_clone.parameters
-
-
-def test_clone_random_state():
```
Comment (angela97lin) [inline evalml/tests/component_tests/test_components.py:445]: Don't need this test specifically anymore.

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 2]
```diff
@@ -1430,39 +1431,19 @@ def test_nonlinear_clone_init(nonlinear_regression_pipeline_class):
...
-        'Linear Regressor': {
-            'fit_intercept': True,
-            'normalize': True,
-        }
-    }
-    pipeline = linear_regression_pipeline_class(parameters=parameters, random_state=42)
-    pipeline_clone = pipeline.clone(random_state=42)
-    assert pipeline_clone.random_state == pipeline.random_state
-
-    pipeline = linear_regression_pipeline_class(parameters=parameters, random_state=2)
```
Comment (angela97lin) [inline evalml/tests/pipeline_tests/test_pipelines.py:1451]: Was duplicated checks? 🤔

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 3]
```diff
@@ -1401,85 +1401,59 @@ class PipelineWithDropCol(BinaryClassificationPipeline):
...
-        }
-    }
-    pipeline = linear_regression_pipeline_class(parameters=parameters)
-    pipeline_clone = pipeline.clone()
-    assert pipeline.parameters == pipeline_clone.parameters
-
-
-def test_nonlinear_clone_init(nonlinear_regression_pipeline_class):
+@pytest.mark.parametrize("is_linear", [True, False])
+def test_clone_init(is_linear, linear_regression_pipeline_class, nonlinear_regression_pipeline_class):
```
Comment (angela97lin) [inline evalml/tests/pipeline_tests/test_pipelines.py:1405]: Combined `test_clone_init ` and `test_nonlinear_clone_init` via parametrize

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 4]
```diff
@@ -1401,85 +1401,59 @@ class PipelineWithDropCol(BinaryClassificationPipeline):
...
-    assert pipeline_clone.random_state == pipeline.random_state
-
-    pipeline = linear_regression_pipeline_class(parameters=parameters, random_state=2)
-    pipeline_clone = pipeline.clone(random_state=2)
-    assert pipeline_clone.random_state == pipeline.random_state
-
-
-def test_clone_fitted(X_y_binary, logistic_regression_binary_pipeline_class):
+@pytest.mark.parametrize("is_linear", [True, False])
+def test_clone_fitted(is_linear, X_y_binary, logistic_regression_binary_pipeline_class, nonlinear_binary_pipeline_class):
```
Comment (angela97lin) [inline evalml/tests/pipeline_tests/test_pipelines.py:1427]: Combined `test_nonlinear_clone_fitted` and `test_clone_fitted` via parametrize

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by chukarsten — COMMENTED** [review]
I really like the subtractions in this PR.  Very well done.

**Pull request merged**